### PR TITLE
G11 create consultation slots

### LINKED ISSUE #312: G.1.1 Create consultation slots

### LINKED ISSUE #313: G.1.2 Edit Consultation Slots

### LINKED ISSUE #314: G.1.3 Delete Consultation Slots

### LINKED ISSUE #315: G.2.1 View Consultation Slots

### LINKED ISSUE #316: G.2.2 Apply for Consultation Slots

### PR DESCRIPTION
**Pull request by ianleck**
## Description:
- All consultation slot functions - create edit delete, view and register/unregister
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/controllers/consultation.controller.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    try {
+      const consultationSlots = await ConsultationService.viewFilteredConsultationSlots(
+        user.accountId,
+        dateStart,
+        dateEnd
+      );
+      return apiResponse.result(
+        res,
+        {
+          message: CONSULTATION_RESPONSE.CONSULTATION_DELETE,
```
Comment (hiixdayah) [inline src/controllers/consultation.controller.ts]: Why is this response CONSULTATION_DELETE

File: src/controllers/consultation.controller.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          consultationSlots,
+        },
+        httpStatusCodes.OK
+      );
+    } catch (e) {
+      logger.error(
+        '[consultationController.viewAllConsultationSlots]:' + e.message
```
Comment (hiixdayah) [inline src/controllers/consultation.controller.ts]: consultationController.viewFilteredConsultationSlots

**Review by y1bs — DISMISSED** [review]
Looks ok on the frontend perspective

File: src/controllers/consultation.controller.ts (hunk 1 above)
Comment (ianleck) [reply src/controllers/consultation.controller.ts]: put wrong

**Review by kengthong — APPROVED** [review]

**Pull request merged**